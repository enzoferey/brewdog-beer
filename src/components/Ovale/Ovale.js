import React from "react";
import PropTypes from "prop-types";
import style from "./ovale.scss";

const Ovale = ({ from, to }) => (
  <svg
    className={style.main}
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 1445.797 391.5"
  >
    <defs>
      <linearGradient
        id="linear-gradient"
        x1="0.061"
        x2="1.479"
        y2="1.666"
        gradientUnits="objectBoundingBox"
      >
        <stop offset="0" stopColor={from} />
        <stop offset="1" stopColor={to} />
      </linearGradient>
    </defs>
    <path
      fill="url(#linear-gradient)"
      d="M722.9,0c399.246,0,722.9,175.281,722.9,391.5H0C0,175.281,323.653,0,722.9,0Z"
      transform="translate(1445.797 391.5) rotate(180)"
    />
  </svg>
);

Ovale.propTypes = {
  from: PropTypes.string,
  to: PropTypes.string,
};

export default Ovale;
