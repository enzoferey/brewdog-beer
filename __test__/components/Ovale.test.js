import React from "react";
import Ovale from "components/Ovale";

const props = {
  from: "white",
  to: "black",
};

test("renders correctly with a gradient from `from` to `to` props", () => {
  testSnapshot(<Ovale {...props} />);
});
