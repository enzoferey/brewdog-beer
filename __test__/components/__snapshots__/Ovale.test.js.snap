// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a gradient from \`from\` to \`to\` props 1`] = `
<svg
  className={undefined}
  viewBox="0 0 1445.797 391.5"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <linearGradient
      gradientUnits="objectBoundingBox"
      id="linear-gradient"
      x1="0.061"
      x2="1.479"
      y2="1.666"
    >
      <stop
        offset="0"
        stopColor="white"
      />
      <stop
        offset="1"
        stopColor="black"
      />
    </linearGradient>
  </defs>
  <path
    d="M722.9,0c399.246,0,722.9,175.281,722.9,391.5H0C0,175.281,323.653,0,722.9,0Z"
    fill="url(#linear-gradient)"
    transform="translate(1445.797 391.5) rotate(180)"
  />
</svg>
`;
